distributed/tests/test_utils_test.py

```diff
@@ -772,3 +773,33 @@ async def test(s):
     with pytest.raises(CustomError):
         test()
     assert test_done
+
+
+def test_popen_write_during_terminate_deadlock():
+    # Fabricate a command which, when terminated, tries to write more than the pipe buffer can hold (65536 bytes).
+    # This would deadlock if `proc.wait()` was called, since the process will be trying to write to stdout, but
+    # stdout isn't being cleared because our process is blocked in `proc.wait()`.
+    # `proc.communicate()` is necessary: https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait
+    with popen(
+        [
+            sys.executable,
+            "-c",
+            "; ".join(
+                [
+                    "import signal",
+                    "import time",
+                    "import threading",
+                    "e = threading.Event()",
+                    "signal.signal(signal.SIGINT, lambda *args: [print('x' * 131072), e.set()])",
+                    # ^ 131072 is 2x the size of the default Linux pipe buffer
+                    "print('ready', flush=True)",
+                    "e.wait()",
+                ]
+            ),
```

Comment:
```suggestion
            textwrap.dedent(
                """
                import signal
                import time
                import threading
                e = threading.Event()
                # ^ 131072 is 2x the size of the default Linux pipe buffer
                signal.signal(signal.SIGINT, lambda *args: [print('x' * 131072), e.set()])
                print('ready', flush=True)
                e.wait()
                """
            ),
```
+ ``import textwrap`` at module top